lib/ansible/modules/net_tools/nsupdate.py

```diff
@@ -378,7 +387,8 @@ def main():
             record=dict(required=True, type='str'),
             type=dict(required=False, default='A', type='str'),
             ttl=dict(required=False, default=3600, type='int'),
-            value=dict(required=False, default=None, type='list')
+            value=dict(required=False, default=None, type='list'),
+            protocol=dict(required=False, default='tcp', choices=protocols, type='str')
```

Comment:
I would skip the `protocols` variable, and just insert the choices list directly. Both because I think it's slightly easier to read, and because most modules does it that way.

`tsig_algs` is a bit of a special case, given the size of that list.